File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
       if(s.size()!=t.size()) return 0;
        int h[26]={0};
        int h2[26]={0};
        for(char c:s)
          h[c-'a']++;
        for(char c:t)
           h2[c-'a']++;
        for(int i=0;i<26;i++)
            if(h[i]!=h2[i])
              return 0;
        return 1; 
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character,Integer>hm = new HashMap<>();
        if(s.length() != t.length()) return false;
        
        for(char ch : s.toCharArray()){
           int preF = hm.getOrDefault(ch,0);
          hm.put(ch,preF+1);   
        }
        for(char ch : t.toCharArray()) {
           int preF = hm.getOrDefault(ch,0);
           if(preF == 0){return false;}
             hm.put(ch,preF-1);
             if(hm.get(ch) == null || hm.get(ch) == 0) {
                   hm.remove(ch);
             }
        }
       return hm.isEmpty();  
    }
}